types/propEq.d.ts

```diff
@@ -1,6 +1,31 @@
-export function propEq<T>(val: T): {
-  <K extends PropertyKey>(name: K): (obj: Record<K, T>) => boolean;
-  <K extends PropertyKey>(name: K, obj: Record<K, T>): boolean;
+import { Placeholder } from 'ramda';
+import { WidenLiterals } from '../util/tools';
+
+export function propEq(__: Placeholder): never;
+export function propEq<const V>(val: V): {
+  <K extends number>(name: K): <U extends any[]>(array: V extends U[K] ? V[] : never) => boolean;
```

Comment:
Ah crap, I missed this when I left that other comment about this MR not being a problem
```
array: V extends U[K] ? V[] : never
```
This is the thing we can't have. It breaks combining with other functions, eg`R.filter(R.propEq('type', 'something'))`. The predicate type signatures don't match because of the `: never`